protocol/wlr-layer-shell-unstable-v1.xml

```diff
@@ -0,0 +1,279 @@
+        request, passing along the serial of the configure event.
+
+        If the client receives multiple configure events before it
+        can respond to one, it only has to ack the last configure event.
+
+        A client is not required to commit immediately after sending
+        an ack_configure request - it may even ack_configure several times
+        before its next surface commit.
+
+        A client may send multiple ack_configure requests before committing, but
+        only the last request sent before a commit indicates which configure
+        event the client really is responding to.
+      </description>
+      <arg name="serial" type="uint" summary="the serial from the configure event"/>
+    </request>
+
+    <request name="destroy" type="destructor">
+      <description summary="destroy the layer_surface">
+        This request destroys the layer surface.
+      </description>
+    </request>
+
+    <event name="configure">
+      <description summary="suggest a surface change">
+        The configure event asks the client to resize its surface.
+
+        Clients should arrange their surface for the new states, and then send
+        an ack_configure request with the serial sent in this configure event at
+        some point before committing the new surface.
+
+        The client is free to dismiss all but the last configure event it
+        received.
+
+        The width and height arguments specify the size of the window in
+        surface-local coordinates.
+
+        The size is a hint, in the sense that the client is free to ignore it if
+        it doesn't resize, pick a smaller size (to satisfy aspect ratio or
+        resize in steps of NxM pixels). If the client picks a smaller size and
+        is anchored to two opposite anchors (e.g. 'top' and 'bottom'), the
+        surface will be centered on this axis.
+
+        If the width or height arguments are zero, it means the client should
+        decide its own window dimension.
+      </description>
+      <arg name="serial" type="uint"/>
+      <arg name="width" type="uint"/>
+      <arg name="height" type="uint"/>
+    </event>
+
+    <enum name="error">
+      <entry name="invalid_input_device" value="0" summary="input device bitfield is invalid"/>
+      <entry name="invalid_anchor" value="1" summary="anchor bitfield is invalid"/>
+      <entry name="invalid_surface_state" value="2"
+	     summary="the client provided an invalid surface state"/>
+    </enum>
+
+    <enum name="anchor" bitfield="true">
+      <entry name="top" value="1" summary="the top edge of the anchor rectangle"/>
+      <entry name="bottom" value="2" summary="the bottom edge of the anchor rectangle"/>
+      <entry name="left" value="4" summary="the left edge of the anchor rectangle"/>
+      <entry name="right" value="8" summary="the right edge of the anchor rectangle"/>
+    </enum>
+  </interface>
+
+  <interface name="zwlr_layer_input_v1" version="1">
```

Comment:
What is this meant to accomplish that cannot be done with the surface input region and how does this interact with surface input region?

Reply:
This is unrelated to the input region. The input region will affect which parts of the surface are elegible to receive pointer/touch events. Strictly speaking you can set the input region to 0x0@0,0 to avoid getting pointer/touch events but this is a fat hack. The layer input device lets you be more explicit about it, and also specifies keyboard and tablet semantics.

Reply:
Note that a region is not a rectangle. There are empty regions that contain no rectangles (and they don't need to contain a 0x0@0,0 rectangle), so it's not a hack, it's legit.

Reply:
Alright, fair enough. However, it's still less explicit and doesn't specify keyboard or tablet behavior.

Reply:
IIRC tablet behavior is defined to respect the wl_surface input region in the tablet protocol. But yeah, you're right, the keyboard is missing.